webapp/publisher/views.py

```diff
@@ -78,16 +80,57 @@ def get_account_details():
         return _handle_errors(api_error)
 
     flask_user = flask.session["openid"]
+
+    subscriptions = None
+    subscribed_to_newsletter = None
+    # don't rely on marketo to show the page,
+    # if anything fails, just continue and don't show
+    # this section
+    try:
+        marketo = marketo_api.MarketoApi()
+        marketo_user = marketo.get_user(flask_user["email"])
+        marketo_subscribed = marketo.get_newsletter_subscription(
+            marketo_user["id"]
+        )
+        subscribed_to_newsletter = False
+        if (
+            "snapcraftnewsletter" in marketo_subscribed
+            and marketo_subscribed["snapcraftnewsletter"]
+        ):
+            subscribed_to_newsletter = True
+    except Exception:
+        pass
+
+    if subscribed_to_newsletter is not None:
```

Comment:
Think you can remove the `is not None`

Reply:
it has 3 states though, None (nothing came back from the server), False (you're not subscribed) and True (you're subscribed). Is there a better way of doing this?

Reply:
You could update `subscriptions` inside the try instead of setting `subscribed_to_newsletter` and just get rid of this `if`?